tests/unit/types/arrays/test_documentarray.py

```diff
@@ -332,6 +332,22 @@ def test_da_sort_by_document_interface_in_proto():
     assert da[0].embedding.shape == (1,)
 
 
+def test_da_sort_topk():
+    da = DocumentArray(
+        [Document(id=i, copy=True, scores={'euclid': 10 - i}) for i in range(10)]
+    )
+    assert da[0].id == '0'
+    assert da[0].scores['euclid'].value == 10
+
+    da.sort(top_k=3, key=lambda d: d.scores['euclid'].value, reverse=True)
```

Comment:
This test should validate the up to top_k things are kept, and the rest are not kept in order.

Also it should validate that the set of Documents is still valid